pkg/kademlia/routing.go

```diff
@@ -202,3 +205,19 @@ func (rt *RoutingTable) GetBucketTimestamp(bIDBytes []byte) (time.Time, error) {
 func (rt *RoutingTable) iterate(opts storage.IterateOptions, f func(it storage.Iterator) error) error {
 	return rt.nodeBucketDB.Iterate(opts, f)
 }
+
+// ConnFailure implements the Transport failure function
+func (rt RoutingTable) ConnFailure(ctx context.Context, node *pb.Node, err error) {
+	err2 := rt.ConnectionFailed(node)
+	if err2 != nil {
+		zap.L().Debug(fmt.Sprintf("error with ConnFailure hook  %+v : %+v", err, err2))
+	}
+}
+
+// ConnSuccess implements the Transport success function
+func (rt RoutingTable) ConnSuccess(ctx context.Context, node *pb.Node) {
+	err := rt.ConnectionSuccess(node)
+	if err != nil {
+		zap.L().Debug("connection success error:", zap.Error(err))
```

Comment:
add logger to struct, call here

Reply:
Here too: Let's keep the receivers consistent and change (rt RoutingTable) to (rt *RoutingTable)

Reply:
That's going to change it to a pointer receiver, which will change the function signature of the Observer interface. I'm not sure I agree it should be changed.

Reply:
It wouldn't change the observer interface.